### DISCUSSION TITLE
How to use comma as decimal marker and period as thousand separator?

### DISCUSSION THREAD

**Discussion by tifandotme** [dream-num/univer#3380] [Q&A | 常见问题]
As per title, are there any capability for it? and how can I achieve it?

Thank you in advance

**Comment by jikkai** [comment]
https://univer.ai/guides/sheet/features/numfmt

**Comment by tifandotme** [comment]
Thank you for the response!

I've tried the plugin, it can specify the number of decimal places, as well as whether to have thousand separator. But it doesn't change the behaviour of commas and periods. That is, when a user inputed a comma, Univer still treats it as thousand separator and period as decimal point. (please tell me if I missed something)

Can I switch the behaviour of commas and period as well as how they are displayed in numeric values?

For reference [Google Sheets uses spreadsheet locales](https://support.google.com/docs/answer/56470?rd=1#zippy=%2Ccustom-number-formatting) to determine this.

<img width="701" alt="Screenshot 2024-09-09 at 14 47 45" src="https://github.com/user-attachments/assets/4759487b-fc9c-44d0-83fd-7c5976de6b42">

**Comment by Gggpound** [comment]
Can you give me an example? What is your input and what is it expected to achieve?

**Reply by tifandotme** [reply]
- Input: 0,582038 -> expect: 0,582038 -> instead: 582,038
- Input: 22,5192933 -> expect: 22,5192933 -> instead: 225,192,933
- Input: 0,2 -> expect: 0,2 -> instead: 2
- Input: 1.100.100 -> expect: 1.100.100 (cell value: 1100100, cell type: number) -> instead: 1.100.100 (cell value: "1.100.100", cell type: string)
